### ISSUE TITLE
Update install instructions

### ISSUE THREAD

**Issue by nsghub** [Klaus3d3/XdripWidgetForAmazfit-Klaus#2]
I hope this makes it clear for everyone who wants to install the apk.

Before you can install the apk you need to unlock the bootloader and gain root access:

1. Unlock the bootloader
Go to https://forum.xda-developers.com/smartwatch/amazfit/tutorial-unlock-bootloader-warning-void-t3654011 and follow the instructions

2. Download temporary bootloader
Download this file https://mega.nz/#!cckXRRbD!aWEEyD28OZUpaVaSshzAHqUQbGntV1wigjt95Tx2BZE and unzip it.

3. Flash temporary bootloader
- With your watch connected to the computer issue the command "adb shell reboot bootloader"
- When the fastboot screen appears on the watch, you can flash the temp bootloader: fastboot boot /home/tobias/src/pace/temp_root_adb/boot-US-adb-root.img 
- You might see a FAILED message from fastboot that can be ignored. The watch will reboot.

4. Pushing the apk
- After the watch finished rebooting you need to remount the file system writable: adb remount
- And finally push the apk onto the watch: adb push xDripwidget.apk /system/priv-app
- Reboot the watch again (root priviliges will be lost): adb reboot

5. Updating the app:
- Repeat steps 3 and 4.

**Comment by Klaus3d3** [comment]
Thanks, i will update the readme this evening

**Comment by nsghub** [comment]
Oh and I think you have enable "Lab mode" first by tapping a couple of times on the serial number. Otherwise adb will report missing permissions.

Another thing regarding the install process. Is it really necessary to push the apk into /system/priv-app? I installed it with `adb install xdrip.apk` and restarted the launcher with `adb shell am force-stop com.huami.watch.launcher`
The widget appears in the list and works just as with the other method. Including the laucher crashes when no data is available ;)
This way you don't have to deal with unlocking and bootloaders.

**Comment by Klaus3d3** [comment]
the crash should be fixed now ;-). and we have a plugin indication like in xdrip itself.

hmm. i copied that from the original developer, who told me otherwise the service on the watch will not run in the background. I'll check that tomorrow. Would make things much easier.

**Comment by nsghub** [comment]
No crash with the latest version. Will test the behaviour tomorrow and report if it updates the values as expected.
Did you change anything else? With the older releases it's possible to sort the widget with [Springboard settings for Pace](https://forum.xda-developers.com/smartwatch/amazfit/dev-create-custom-home-screen-pages-pace-t3751731) but with the latest release it is not listed in the settings anymore.

**Comment by Klaus3d3** [comment]
No, nothing that i know of.

The spring board settings for pace never worked for me. When i changed the position of the widget, it still jumped back after some time. Could'nt find out what causes this. So i gave up cause useless. And i read that it's normal. 

But i'm on Stratos. Can't test with pace.

**Comment by Klaus3d3** [comment]
Ok, it seems as i found out when the rearrangment of the springboards happen.

When i switch to the xDrip widget and let the watch go to powersaving mode u just need to press the power button and ur back on.
But when the watch looses the connection from the phone and then reconnects, its back at the center position. 

At night, when the phone is on my night stand and therefore always in range, it stays there forever.

At these events also the rearrangement to normal happens if changed with spingboard settings before. 

This tells me, that the Amazfit companion app (where u also can change the arrangement) sends the layout again to the watch when it reconnects.

I'm actually not running the Amazfit app but Amazmod app by edotassi, wich is a modified app.
I could ask the developer if he can modifiy this point.

**Comment by Klaus3d3** [comment]
as you might mentioned in the gitter chat, normal installation via adb or HUAMI_Amazfit_tool does not work and causes the launcher to chrash.

**Comment by nsghub** [comment]
Observations so far:

- Installed the app via adb install
- Initial sync has to be made manually via xDrip three dot menu "send values to smart watch".
- ~~Latest version does not crash the launcher if no initial value is available.~~ Still crashes. After rebooting it sometimes works.
- Latest version does not show up in springboard settings. I will check this tonight, because the version that showed up is one that I compiled myself (based on your ~~first~~ commit bbceb29d00699e73e76b397406ffc263e59653c9). I found out, that the debug version is available in springboard settings, while the release version is not.
- As you said, the springboard settings are overwritten by the Amazfit app.

Is the Amazmod app able to fully replace the Amazfit app? Including starting/setup of the watch?
